src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -643,6 +675,85 @@ ReactDOMComponent.Mixin = {
   },
 
   /**
+   * When walking the DOM tree to see if server-rendered markup mismatches
+   * the rendered component, this method checks to make sure the attributes
+   * values of the DOM node match the props of this component.
+   *
+   * This method has side effects because events get registered and the style
+   * attribute gets cached in this._previousStyleCopy.
+   *
+   * @private
+   * @param {DOMElement} node with attributes to test
+   * @param {object} component props to compare to node attributes
+   * @throws when they don't match.
+   */
+  _compareNodeAttributesToProps: function(node, props, transaction) {
+    var attrs = node.attributes;
+    for (var i = 0; i < attrs.length; i++) {
+      var attrName = attrs[i].name;
+      var attrValue = attrs[i].value;
+      if (attrName === DOMProperty.ID_ATTRIBUTE_NAME ||
+        attrName === DOMProperty.ROOT_ATTRIBUTE_NAME ||
+        attrName === ReactMarkupChecksum.CHECKSUM_ATTR_NAME) {
+        continue;
+      }
+      if (attrName === 'class') {
```

Comment:
Thanks for the feedback, @donabrams! 

I realized this yesterday while I was working on the standalone renderer, but I haven't gotten around to patching it. There's also some other cases, like xlinkHref/xlink:href and xmlLang/xml:lang. I'll certainly fix it up with a unit test and a patch if we decide to move forward with this PR, and thanks again!